docs/python_binding.rst

```diff
@@ -117,9 +117,30 @@ To get a python dictionary of the settings, call ``session::get_settings``.
 
 .. _`library reference`: reference.html
 
-Retrieving session statistics in Python is more convenient than that in C++.
-The statistics are stored as an array in ``session_stats_alert``, which will be posted after calling ``post_session_stats()`` in the ``session`` object.
-In order to interpret the statistics array, in C++ it is required to call ``session_stats_metrics()`` to get the indices of these metrics, while in Python it can be done using ``session_stats_alert.values["NAME_OF_METRIC"]``, where ``NAME_OF_METRIC`` is the name of a metric.
+Retrieving session statistics in Python is more convenient than that in C++. The
+statistics are stored as an array in ``session_stats_alert``, which will be
+posted after calling ``post_session_stats()`` in the ``session`` object. In
+order to interpret the statistics array, in C++ it is required to call
+``session_stats_metrics()`` to get the indices of these metrics, while in Python
+it can be done using ``session_stats_alert.values["NAME_OF_METRIC"]``, where
+``NAME_OF_METRIC`` is the name of a metric.
+
+set_alert_notify
+================
+
+The ``set_alert_notify()`` function is not compatible with python. Since it
+requires locking the GIL from within the libtorrent thread, to call the callback,
+it can cause a deadlock with the main thread.
+
+Instead, use the python-specific ``set_alert_fd()`` which takes a file descriptor
```

Comment:
I'd mention that the file descriptor must be non-blockable.

Reply:
I don't think it has to be non-blocking. but not reading from it at sufficient frequency will block libtorrent's internal thread. Making it non-blocking won't really solve that problem, it will just make you miss events instead

Reply:
> I don't think it has to be non-blocking. but not reading from it at sufficient frequency will block libtorrent's internal thread.

That's true. "Pass a blocking FD at your own risk" :)

> Making it non-blocking won't really solve that problem, it will just make you miss events instead

Not really. If the fd is full, it means that the other end is already in "notified" state.

Reply:
good point